example/src/App.js

```diff
@@ -0,0 +1,173 @@
+import React from 'react';
+import MapboxGL from 'react-native-mapbox-gl';
+
+import {
+  View,
+  Text,
+  FlatList,
+  StyleSheet,
+  Modal,
+  Platform,
+  TouchableOpacity,
+  NativeModules
+} from 'react-native';
+
+import { Icon } from 'react-native-elements';
+
+// components
+import MapHeader from './components/common/MapHeader';
+
+// styles
+import sheet from './styles/sheet';
+import colors from './styles/colors';
+
+// utils
+import { IS_ANDROID } from './utils';
+import config from './utils/config';
+
+// examples
+import ShowMap from './components/ShowMap';
+import SetPitch from './components/SetPitch';
+import SetBearing from './components/SetBearing';
+import ShowClick from './components/ShowClick';
+import FlyTo from './components/FlyTo';
+import FitBounds from './components/FitBounds';
+import SetUserTrackingModes from './components/SetUserTrackingModes';
+import ShowRegionDidChange from './components/ShowRegionDidChange';
+
+const styles = StyleSheet.create({
+  header: {
+    marginTop: 48,
+    fontSize: 24,
+    textAlign: 'center',
+  },
+  exampleList: {
+    flex: 1,
+    marginTop: 60 + 12 // header + list padding,
+  },
+  exampleListItemBorder: {
+    borderBottomWidth: StyleSheet.hairlineWidth,
+    borderBottomColor: '#ccc',
+  },
+  exampleListItem: {
+    paddingVertical: 32,
+    paddingHorizontal: 16,
+    flexDirection: 'row',
+    justifyContent: 'space-between',
+  },
+  exampleListLabel: {
+    fontSize: 18,
+  },
+  exampleBackground: {
+    flex: 1,
+    backgroundColor: colors.primary.pinkFaint,
+  },
+})
+
+class ExampleItem {
+  constructor (label, Component) {
+    this.label = label;
+    this.Component = Component;
+  }
+}
+
+const Examples = [
+  new ExampleItem('Show Map', ShowMap),
+  new ExampleItem('Set Pitch', SetPitch),
+  new ExampleItem('Set Bearing', SetBearing),
+  new ExampleItem('Show Click', ShowClick),
+  new ExampleItem('Fly To', FlyTo),
+  new ExampleItem('Fit Bounds', FitBounds),
+  new ExampleItem('Set User Tracking Modes', SetUserTrackingModes),
+  new ExampleItem('Show Region Did Change', ShowRegionDidChange),
+];
+
+class App extends React.Component {
+  constructor (props) {
+    super(props);
+
+    this.state = {
+      activeExample: -1,
+    };
+
+    this.renderItem = this.renderItem.bind(this);
+    this.onCloseExample = this.onCloseExample.bind(this);
+  }
+
+  async componentWillMount () {
+    if (IS_ANDROID) {
+      await MapboxGL.requestPermissions();
+    }
+    MapboxGL.setAccessToken(config.get('accessToken'));
```

Comment:
I don't think we should be dictating to developers how they should be configuring private/config variables. I think inserting the string `INSERT MAPBOX ACCESS TOKEN HERE. Your access token can be found here: Link` should be enough.

Reply:
My only concern having it as a hardcoded string would be that every time someone wants to add an example they would have to add the token here and remove the token here before making a commit.